### PR TITLE
Remove IFromIndexOperation completely

### PR DESCRIPTION
**Pull request by agocke**
We previously decided to remove IFromIndexOperation in favor of using an IUnaryOperation to represent the `^` syntax in C#, but did not have time to remove the interface completely. This PR finishes the change.

### CODE REVIEW

File: src/Compilers/Core/Portable/Operations/UnaryOperatorKind.cs [hunk 1]
```diff
@@ -41,6 +41,11 @@ public enum UnaryOperatorKind
         /// Represents the C# 'false' operator and VB 'IsFalse' operator.
         /// </summary>
         False = 0x6,
+
+        /// <summary>
+        /// Represents the C# '^' operator.
+        /// </summary>
+        Hat = 0x7,
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Operations/UnaryOperatorKind.cs]: >Hat [](start = 8, length = 3)

Is this how it will be called in the language spec?

File: src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -5,8 +5,8 @@ Microsoft.CodeAnalysis.CSharp.CSharpCompilationOptions.CSharpCompilationOptions(
 Microsoft.CodeAnalysis.CSharp.CSharpCompilationOptions.NullableContextOptions.get -> Microsoft.CodeAnalysis.CSharp.NullableContextOptions
 Microsoft.CodeAnalysis.CSharp.CSharpCompilationOptions.WithNullableContextOptions(Microsoft.CodeAnalysis.CSharp.NullableContextOptions options) -> Microsoft.CodeAnalysis.CSharp.CSharpCompilationOptions
 Microsoft.CodeAnalysis.CSharp.LanguageVersion.CSharp8 = 800 -> Microsoft.CodeAnalysis.CSharp.LanguageVersion
-Microsoft.CodeAnalysis.CSharp.LanguageVersion.Preview = 2147483646 -> Microsoft.CodeAnalysis.CSharp.LanguageVersion
 Microsoft.CodeAnalysis.CSharp.LanguageVersion.LatestMajor = 2147483645 -> Microsoft.CodeAnalysis.CSharp.LanguageVersion
+Microsoft.CodeAnalysis.CSharp.LanguageVersion.Preview = 2147483646 -> Microsoft.CodeAnalysis.CSharp.LanguageVersion
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt]: Are there meaningful changes in this file? Please revert changes that simply move lines around

File: src/Workspaces/Core/Portable/PublicAPI.Unshipped.txt [hunk 3]
```diff
@@ -1,4 +1,3 @@
-Microsoft.CodeAnalysis.Editing.DeclarationModifiers.WithIsRef(bool isRef) -> Microsoft.CodeAnalysis.Editing.DeclarationModifiers
```
Comment (AlekseyTs) [inline src/Workspaces/Core/Portable/PublicAPI.Unshipped.txt]: Please revert this change.

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 2)

File: src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt (hunk 2 above)
Comment (agocke) [reply src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt]: Every change to the PublicAPI files is supposed to be followed by a call to `sort-unshipped.cmd`. That's all I did here.

File: src/Compilers/Core/Portable/Operations/UnaryOperatorKind.cs (hunk 1 above)
Comment (agocke) [reply src/Compilers/Core/Portable/Operations/UnaryOperatorKind.cs]: Yup, we decided that in https://github.com/dotnet/csharplang/blob/6904c200e917967c5a06ebea7493b2dea848c4ab/meetings/2018/LDM-2018-09-05.md

File: src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt (hunk 2 above)
Comment (agocke) [reply src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt]: Would you prefer I change that in a separate PR? Or as separate commit to make reviewing easier?

Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt]: > Every change to the PublicAPI files is supposed to be followed by a call to sort-unshipped.cmd. 

This is the first time I hear about that. I see very little benefit and a lot of negatives in doing this. People should be able to examine the diff and quickly see what has changed, not just in PRs, but in history as well. Merge tools should be able to merge non-conflicting API changes automatically.

>Would you prefer I change that in a separate PR? Or as separate commit to make reviewing easier?

I prefer that sorting was not done in this PR, or any other PR.

---
In reply to: [267972689](https://github.com/dotnet/roslyn/pull/34306#discussion_r267972689) [](ancestors = 267972689)

Comment (agocke) [reply src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt]: Sorting is extremely helpful for resolving conflicts in merge PRs, as changes to the public API are extremely likely to be added in the same section without sorting.

Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt]: > Sorting is extremely helpful for resolving conflicts in merge PRs, as changes to the public API are extremely likely to be added in the same section without sorting.

I have an opposite experience. Also, when one merges additions from different branches he should be able to take both sides and move on, without having to run some other tools to sort, etc. 

---
In reply to: [268290574](https://github.com/dotnet/roslyn/pull/34306#discussion_r268290574) [](ancestors = 268290574)

**Comment by agocke** [comment]
@AlekseyTs I've removed the formatting change and just left the required changes.
